src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -871,8 +872,12 @@ public boolean process(Person person, long time) {
         } else {
           type = EncounterType.fromString(encounterClass);
         }
+        String desiredSpeciality = ClinicianSpecialty.GENERAL_PRACTICE;
+        if (this.specialty != null && !this.specialty.isEmpty()) {
+          desiredSpeciality = this.specialty;
```

Comment:
Do we want to make sure that `this.speciality` exists within the array of `ClinicianSpeciality.getSpecialities()`?

Reply:
Now throws an `IllegalArgumentException` if there is a specialty that it doesn't recognize.